### PR TITLE
Mkarnowski mtls

### PR DESCRIPTION
**Pull request by mkarnowski**
initial commit for a simplified guide for mtls.

### CODE REVIEW

File: docs/guides/using-tls-mutual-authentication.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+---
+title: Using Mutual TLS Authentication
```
Comment (russorat) [inline docs/guides/using-tls-mutual-authentication.md:2]:
```suggestion
title: Using Mutual TLS (mTLS) Authentication
```

File: docs/guides/using-tls-mutual-authentication.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+description: Protecting your endpoints with mutual TLS authentication (mTLS)
+tags:
+  - guides
+  - mtls
+  - tls
+  - mutual authentication
+  - security
+---
+
+Mutual TLS Authenctication (mTLS) is a method for providing security to sensitive applications in a zero trust security posture.  This additional layer of security provides confidence by validating the identify of both the client and server before any transfer of data.  In a majority of the TLS connections being made everyday, the client will verify the identify of server confirming that the correct server is being accessed.  For example, when accessing [https://ngrok.com/docs](https://ngrok.com/docs), you will receive a certificate correctly identifying the server as ngrok.com.
```
Comment (russorat) [inline docs/guides/using-tls-mutual-authentication.md:12]:
```suggestion
Mutual TLS Authenctication (mTLS) is a method for providing security to sensitive applications in a zero trust security posture.  This additional layer of security provides confidence by validating the identify of both the client and server before any transfer of data.  In a majority of the TLS connections being made everyday, the client will verify the identify of server confirming that the correct server is being accessed.  For example, when accessing https://ngrok.com/docs, you will receive a certificate correctly identifying the server as ngrok.com.  
```

File: docs/guides/using-tls-mutual-authentication.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+mTLS adds an additional flow to the previous TLS server verification steps.  The client will still verifiy the server's identity, but now the server will in turn also verify the identity of the client.  By verifying the clients, the server owner is able to restrict access only to verified clients strengthening security.
+![](img/mtls-diag.png)
+
+This document will provide a quick start guide for how to enable mTLS on your ngrok hosted endpoints.  For a deeper understanding for how mTLS is implemented within ngrok, reference the [Mutual TLS module page.](https://ngrok.com/docs/cloud-edge/modules/mutual-tls/)
```
Comment (russorat) [inline docs/guides/using-tls-mutual-authentication.md:17]:
```suggestion
This document will provide a quick start guide for how to enable mTLS on your ngrok hosted endpoints. For a deeper understanding for how mTLS is implemented within ngrok, reference the [Mutual TLS module page](https://ngrok.com/docs/cloud-edge/modules/mutual-tls/).
```

File: docs/guides/using-tls-mutual-authentication.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+
+## **Prerequisites**
+
+A certificate authority (CA) is required for mTLS.  The CA is responsible for issuing and digitally signing certificates (client certificates).  The CA will also be used to verify the authenticity of the certificates.
```
Comment (russorat) [inline docs/guides/using-tls-mutual-authentication.md:22]: should we mention that the CA is uploaded to ngrok?

File: docs/guides/using-tls-mutual-authentication.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+This document will provide a quick start guide for how to enable mTLS on your ngrok hosted endpoints.  For a deeper understanding for how mTLS is implemented within ngrok, reference the [Mutual TLS module page.](https://ngrok.com/docs/cloud-edge/modules/mutual-tls/)
+
+
+## **Prerequisites**
+
+A certificate authority (CA) is required for mTLS.  The CA is responsible for issuing and digitally signing certificates (client certificates).  The CA will also be used to verify the authenticity of the certificates.  
+
+1. CA certificate to be used by ngrok to verify the clients
+2. Client certificates signed by the CA used to access the endpoints
```
Comment (russorat) [inline docs/guides/using-tls-mutual-authentication.md:25]: where do these live? who puts them there and when?

File: docs/guides/using-tls-mutual-authentication.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+   x509 -req -days 365 -sha256 -in  ngrokclient.csr \
+   -CA ca.crt -CAkey ca.key -set_serial 0x"$(openssl rand -hex 16)" -out ngrokclient.crt
+```
+
+
+## **Enable mTLS**
+With the prerequisites being met, let's see how simple it is to enable mTLS with ngrok.  Examples for using either the Agent CLI or ngrok Cloud Edges are provided below.
+
+### Agent CLI
+Specify the CA certificate when starting the tunnel.<br></br>
```
Comment (russorat) [inline docs/guides/using-tls-mutual-authentication.md:66]: this seems a little short. maybe add a paragraph below as to what this command is doing? for example, 'this command uploads the CA to the ngrok saas platform and then uses it to verify requests?'

**Review by samcrichard — APPROVED** [review]

**Pull request merged**